BetterPasswordPlugin.inc.php

```diff
@@ -208,17 +199,109 @@ function checkPassword($hookName, $args) {
 			}
 		}
 	}
-
+        /**
+	 * Check for newly added blacklists. If ran the first time creates the list of blacklists
+	 * @return 0|1|2 0 if update failed, 1 if setting is updated, 2 if new setting has been created
+	 */
+        
+        function blacklistSetting() {
+            $prevBlacklist = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+            $updateSettings = false;
+            $newBlacklist = array();
+            foreach ($this->getBlacklists() as $filename) {
+                    $newBlacklist[$filename] = sha1_file($filename);
+            }
+            if (is_null($prevBlacklist)) {
+                $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $newBlacklist);
+                $updateSettings = true;
+            }
+            else {
+                if ($prevBlacklist != $newBlacklist) {
+                    $updateTempFile = $this->handleTempFile(false);
+                    if($updateTempFile) {
+                        $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $blacklistFileSettingHash);
+                        $updateSettings = true;
+                    }
+                    else {
+                        return $updateSettings;
+                    }
+                }
+            }
+            return $updateSettings;
+        }
+        
+        /**
+         * Creates a temporary file to aggregate all the passwords from the blacklist files if the temporary file doesn't exist
+         * @return array The file path for the temporary file
+         */
+        function getTempFile() {
+            import('lib.pkp.classes.file.PrivateFileManager');
+            $fileMgr = new PrivateFileManager();
+            $tempFileDir = realpath($fileMgr->getBasePath()) . DIRECTORY_SEPARATOR . 'betterPassword';
+            if (!$fileMgr->fileExists($tempFileDir, 'dir')) {
+			$success = $fileMgr->mkdirtree($tempFileDir);
+			if (!$success) {
+                            // Files directory wrong configuration?
```

Comment:
One of the "benefits" of the `assert()` was the side effect that it would write to the error log.  [`FileManager::mkdirtree()`](https://github.com/pkp/pkp-lib/blob/a99782f787362879abd0492f53f2ca5166e721d2/classes/file/FileManager.inc.php#L348-L367) doesn't fully report errors encountered, so we should add an `error_log()` statement here to log a message for the sysadmin.